File: backend/src/main/java/com/mortl/dancenetwork/model/Event.java
package com.mortl.dancenetwork.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Entity
@Table(name = "event")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {

  @Id
  @GeneratedValue
  @Column(nullable = false)
  private Long id;

  @Column(nullable = false)
  private UUID creator;

  @Column(nullable = false)
  @NonNull
  private LocalDate startDate;

  private LocalDate endDate;

  @Column(nullable = false)
  @NonNull
  private String name;

  private String location;

  private String website;

  private String email;

  private boolean published;
}

File: dance_network_frontend/lib/model/event.dart
class Event {
  final int eventId;
  final String creator;
  final String profileImageUrl;
  final String bannerImageUrl;
  final String startDate;
  final String? endDate;
  final String startTime;
  final String title;
  final String location;
  final String website;
  final String email;

  Event({
    required this.eventId,
    required this.creator,
    required this.profileImageUrl,
    required this.bannerImageUrl,
    required this.startDate,
    required this.endDate,
    required this.startTime,
    required this.title,
    required this.location,
    required this.website,
    required this.email,
  });

  factory Event.fromMap(Map<String, dynamic> event) {
    return Event(
      eventId: event['id'] ?? 0,
      creator: event['creator'] ?? "",
      profileImageUrl: event['profileImage'] ?? "",
      bannerImageUrl:  event['bannerImage'] ?? "",
      startDate: event['startDate'] ?? "Unknown date",
      endDate: event['endDate'],
      startTime: event['startTime'] ?? "Unknown time",
      title: event['name'] ?? "",
      location: event['location'] ?? "",
      website: event['website'] ?? "",
      email: event['email'] ?? "",
    );
  }
}
